libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Rotoye.cpp

```diff
@@ -0,0 +1,162 @@
+ * #define BATTMONITOR_SMBUS_Rotoye_CHARGE_STATUS         0x0d    // relative state of charge
+ * #define BATTMONITOR_SMBUS_Rotoye_BATTERY_STATUS        0x16    // battery status register including alarms
+ * #define BATTMONITOR_SMBUS_Rotoye_BATTERY_CYCLE_COUNT   0x17    // cycle count
+ * #define BATTMONITOR_SMBUS_Rotoye_DESIGN_VOLTAGE        0x19    // design voltage register
+ * #define BATTMONITOR_SMBUS_Rotoye_MANUFACTURE_DATE      0x1b    // manufacturer date
+ * #define BATTMONITOR_SMBUS_Rotoye_SERIALNUM             0x1c    // serial number register
+ * #define BATTMONITOR_SMBUS_Rotoye_HEALTH_STATUS         0x4f    // state of health
+ * #define BATTMONITOR_SMBUS_Rotoye_SAFETY_ALERT          0x50    // safety alert
+ * #define BATTMONITOR_SMBUS_Rotoye_SAFETY_STATUS         0x51    // safety status
+ * #define BATTMONITOR_SMBUS_Rotoye_PF_ALERT              0x52    // safety status
+ * #define BATTMONITOR_SMBUS_Rotoye_PF_STATUS             0x53    // safety status
+*/
+
+// Constructor
+AP_BattMonitor_SMBus_Rotoye::AP_BattMonitor_SMBus_Rotoye(AP_BattMonitor &mon,
+                                                   AP_BattMonitor::BattMonitor_State &mon_state,
+                                                   AP_BattMonitor_Params &params,
+                                                   AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev)
+    : AP_BattMonitor_SMBus(mon, mon_state, params, std::move(dev))
+{}
+
+void AP_BattMonitor_SMBus_Rotoye::timer()
+{
+	// check if PEC is supported
+    if (!check_pec_support()) {
+        return;
+    }
+
+    uint16_t data;
+    uint32_t tnow = AP_HAL::micros();
+
+    // read voltage (V)
+    if (read_word(BATTMONITOR_SMBUS_VOLTAGE, data)) {
+        _state.voltage = (float)data * 0.001f;
+        _state.last_time_micros = tnow;
+        _state.healthy = true;
+    }
+
+    // read cell voltages
+    for (uint8_t i = 0; i < BATTMONITOR_SMBUS_Rotoye_NUM_CELLS; i++) {
+        if (read_word(Rotoye_cell_ids[i], data)) {
+            _has_cell_voltages = true;
+            _state.cell_voltages.cells[i] = data;
+        } else {
+            _state.cell_voltages.cells[i] = UINT16_MAX;
+        }
+    }
+
+    // timeout after 5 seconds
+    if ((tnow - _state.last_time_micros) > AP_BATTMONITOR_SMBUS_TIMEOUT_MICROS) {
+        _state.healthy = false;
+        return;
+    }
+
+    // read current (A)
+    if (read_word(BATTMONITOR_SMBUS_CURRENT, data)) {
+        _state.current_amps = -(float)((int16_t)data) * 0.001f;
+        _state.last_time_micros = tnow;
+    }
+
+    read_full_charge_capacity();
+
+    // FIXME: Perform current integration if the remaining capacity can't be requested
+    read_remaining_capacity();
+
+    read_temp();
+
+    read_serial_number();
+}
+
+// read_block - returns number of characters read if successful, zero if unsuccessful
+uint8_t AP_BattMonitor_SMBus_Rotoye::read_block(uint8_t reg, uint8_t* data, bool append_zero) const
```

Comment:
Once `check_pec_support` is removed I'm pretty sure you can remove all the `read_block` stuff as it is now unused.

Reply:
I've commented this function out since we're not using it now

Reply:
Just remove it, no need to keep it around commented, it can easily be grabbed if needed again.